src/rln_contract.ts

```diff
@@ -0,0 +1,91 @@
+import { ethers } from "ethers";
+
+import { RLN_ABI } from "./const.js";
+import { RLNInstance } from "./rln.js";
+
+type Member = {
+  pubkey: string;
+  index: number;
+};
+
+export class RLNContract {
+  private contract: ethers.Contract;
+  private membersFilter: ethers.EventFilter;
+
+  private members: Member[] = [];
+
+  constructor(
+    address: string,
+    provider: ethers.Signer | ethers.providers.Provider
+  ) {
+    this.contract = new ethers.Contract(address, RLN_ABI, provider);
+    this.membersFilter = this.contract.filters.MemberRegistered();
+  }
+
+  public getContract(): ethers.Contract {
```

Comment:
```suggestion
  public get contract(): ethers.Contract {
```

Why not using a getter ? makes it easier to use IMHO.